StokesProblem.hh

```diff
@@ -0,0 +1,231 @@
+#include <deal.II/base/function.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/synchronous_iterator.h>
+#include <deal.II/base/work_stream.h>
+
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/full_matrix.h>
+
+#include <deal.II/lac/block_sparse_matrix.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/constraint_matrix.h>
+#include <deal.II/lac/precondition_block.h>
+
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/compressed_sparsity_pattern.h>
+#include <deal.II/lac/sparse_ilu.h>
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/grid/tria_boundary_lib.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_out.h>
+
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_nothing.h>
+
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/solution_transfer.h>
+
+#include <deal.II/lac/compressed_set_sparsity_pattern.h>
+#include <deal.II/lac/sparse_direct.h>
+#include <deal.II/hp/dof_handler.h>
+#include <deal.II/hp/fe_values.h>
+
+#include <deal.II/base/std_cxx1x/bind.h>
+
+#include "CopyData.hh"
+#include "ExactSolutionEx1.hh"
+#include "ExactSolutionEx2.hh"
+#include "ExactSolutionEx3.hh"
+#include "ExactSolutionEx4.hh"
+#include "Parameters.hh"
+#include "RightHandSideEx1.hh"
+#include "RightHandSideEx3.hh"
+#include "ScratchData.hh"
+
+using namespace dealii;
+
+template <int dim>
+class StokesProblem
+{
+public:
+  typedef typename hp::DoFHandler<dim>::active_cell_iterator DoFHandler_active_cell_iterator;
+  typedef typename hp::DoFHandler<dim>::cell_iterator DoFHandler_cell_iterator;
+  typedef typename Triangulation<dim>::active_cell_iterator Triangulation_active_cell_iterator;
+  typedef typename Triangulation<dim>::cell_iterator Triangulation_cell_iterator;
+
+  StokesProblem (bool verbose, EXAMPLE example, QUADRATURE quadrature,
+                 REFINEMENT refinement, unsigned int max_degree);
+
+  ~StokesProblem();
+
+  void generate_mesh();
+
+
+  void setup_system();
+
+  void assemble_system();
+
+  void solve();
+
+  double pressure_mean_value () const;
+
+  void compute_error();
+
+  void estimate_error();
+
+  void mark_cells(const unsigned int cycle, const double theta);
+
+  void output_results (const unsigned int cycle);
+
+  void refine_in_h_p();
+
+  unsigned int n_active_cells();
+
+  types::global_dof_index n_dofs();
+
+  double error_l2_norm();
+
+  double error_estimate_l2_norm();
```

Comment:
I don't understand the logic behind the ordering of the functions. It seems randoms and there are no comments.